@here/harp-mapview/lib/ScreenCollisions.ts

```diff
@@ -67,33 +89,142 @@ export class ScreenCollisions {
     /**
      * Marks the region of the screen intersecting with the given bounding box as allocated.
      *
-     * @param bounds The bounding box in world coordinates.
+     * @param bounds The bounding box in NDC scaled coordinates (i.e. top left is -width/2,
+     * -height/2)
      */
     allocate(bounds: Math2D.Box): void {
         const bbox = {
             minX: bounds.x,
             minY: bounds.y,
             maxX: bounds.x + bounds.w,
-            maxY: bounds.y + bounds.h
+            maxY: bounds.y + bounds.h,
+            type: "box"
         };
         this.rtree.insert(bbox);
     }
 
+    /**
+     * Marks the region of the screen intersecting with the given bounding box as allocated.
+     *
+     * @param bounds The bounding box in screen coordinates.
+     */
+    allocateScreenSpace(bounds: Math2D.Box): void {
+        const shiftedBounds = this.toScreenBoundsSpace(bounds, "box");
+        this.rtree.insert(shiftedBounds);
+    }
+
+    /**
+     * Marks a region of the screen intersecting with the given IBox as allocated.
+     *
+     * @param bounds The bounding box in screen coordinates.
+     * @param isScreenSpace Whether the supplied IBox is in screen coordinates, i.e. 0,0 is top
+     * left, and width,height is bottom right, or NDC Scaled coordiantes, i.e. -width/2, -height/2
+     * is bottom left and width/2 and height/2 is top right (i.e. flipped).
+     */
+    allocateIBox(bounds: IBox, isScreenSpace: boolean): void {
+        const box = {};
+        // Make sure to copy across any extra things.
+        Object.assign(box, bounds);
+        // Copy across the NDC scaled space values required for the rtree
+        if (isScreenSpace) {
+            Object.assign(box, this.toScreenBoundsSpaceFromIBox(bounds));
+        }
+        // Convert box from screen space to screenBounds space
+        this.rtree.insert(box);
+    }
+
     /**
      * Checks if the given bounding box is already allocated.
      *
      * @param bounds The bounding box in world coordinates.
      */
     isAllocated(bounds: Math2D.Box): boolean {
-        const bbox = {
+        const bbox: CollisionBox = {
             minX: bounds.x,
             minY: bounds.y,
             maxX: bounds.x + bounds.w,
-            maxY: bounds.y + bounds.h
+            maxY: bounds.y + bounds.h,
+            type: "box"
         };
 
-        // Re-use array to reduce allocations.
-        return this.rtree.collides(bbox);
+        const results = this.rtree.search(bbox);
```

Comment:
Please measure perf impact of this. We do much more intersection checks now. Before we stopped after the first intersection and now we find all of them. Also a lot of temporary objects are created with this change.

Reply:
Will do.

What do you mean that a lot of temporary objects are created with this change?

Reply:
I compared collides with search over berlin, collide seems to be quite a bit slower, chrome shows 0.4% of the total time, but with the search it shows 0.0% of the total time.